README.md

```diff
@@ -1 +1,145 @@
+
+
+## Overview
+
+Most of web developers are familiar with HTTP headers. They are part of the HTTP requests and responses. Simply put, they provide meta information about the request or response such as content type, content length, language, etc. Without headers, clients including browsers, will be clueless on what the content in the body of the response to expect. They need this meta data to render the content more intelligently. Similarly, without headers, servers will be left guessing what is the nature of the request and what is expected of the server in terms of the response. 
+
+For example, if we have the type of the content as plain text but really it's an HTML code that we want to browsers to render, some browsers might just output HTML code instead of the web page. Not the result we want!
+
+Headers often contain authentication information as well. This enables user sessions.
+
+This lesson will cover how to read any the request headers and write any response headers on the server using the examples of Accept-Charset, and Content-Type, Content-Length headers.
+
+## Request Headers
+
+`request` object of the request handler has an object `headers`. It has names of the headers as keys/properties. The names are lower-cased. For example, if you want to get the value of the `Accept-Charset` request header, you can access `request.headers['accept-charset']`:
+
+```js
+console.log(request.headers['accept-charset'])
+```
+
+To send the request headers, we like to use curl:
+
+```
+curl --header "Accept-Charset: base64" localhost:3000
+```
+
+You can even get CURL for Windows [here](http://www.confusedbycode.com/curl).
+
+Note: If you prefer GUI (graphical user interface) over CLI (command-line interface), you can use the [Postman Chrome app](http://www.getpostman.com) to send a request with a custom `Accept-Charset` header along with its Interceptor extension.
+
+![](postman.png)
+
+Request headers can be arbitrary like "Azat-Header".  So this code will work as well: 
+
+```
+console.log(request.headers['azat-header'])
+```
+
+
+```
+curl --header "Azat-Header: base64" localhost:3000
+```
+
+
+## Response Headers
+
+When it comes to sending back the response, it's a good idea to write response headers. As mentioned before, clients will work more intelligently without content when our servers provide proper response headers. For the same reason, we also want to send back the appropriate status code. 
+
+There are four categories of status codes:
+
+* 2XX: for successfully processed requests
+* 3XX: for redirections or cache information
+* 4XX: for client-side errors
+* 5XX: for server-side errors
+
+Using the core `http` module in the request handler, we can write the header with `response.writeHead(statusCode[, statusMessage][, headers]))` where `headers` is an object. For example, if we want to you send back status 200 with the message okay and two headers `Content-Type` and `Content-Length`, then we can use these statements:
```

Comment:
`[]` means optional. I added more instructions. I'm surprised students don't know this already. MDN uses it a lot as well as almost all official documentation.